extensions/amp-ad-network-doubleclick-impl/0.1/safeframe-host.js

```diff
@@ -544,10 +559,38 @@ export class SafeframeHostApi {
       this.resizeSafeframe(height, width, !!optIsCollapse, messageType);
     } else {
       this.resizeAmpAdAndSafeframe(
-          height, width, messageType, optIsCollapse);
+          height, width, messageType);
     }
   }
 
+  /**
+   * @param {!JsonObject} payload
+   * @private
+   */
+  handleShrinkRequest_(payload) {
+    if (!this.isRegistered_) {
+      return;
+    }
+    if (isNaN(payload['shrink_b']) || isNaN(payload['shrink_t']) ||
+        isNaN(payload['shrink_r']) || isNaN(payload['shrink_l'])) {
+      dev().error(TAG, 'Invalid shrink values.');
+      return;
+    }
+    const shrinkHeight = Number(this.iframe_.height) -
+          payload['shrink_b'] + payload['shrink_t'];
+    const shrinkWidth = Number(this.iframe_.width) -
+          payload['shrink_r'] + payload['shrink_l'];
+    // Make sure we are actually shrinking here.
+    if (shrinkWidth > this.creativeSize_.width ||
```

Comment:
You can do the nan checks here as number + NaN = NaN. E.g:

if (isNaN(shrinkWidth) || isNaN(shrinkHeight) || ..) return;

Reply:
done